src/main/java/reposense/authorship/model/AuthorshipSummary.java

```diff
@@ -1,36 +1,57 @@
 package reposense.authorship.model;
 
 import java.util.HashMap;
+import java.util.LinkedHashMap;
 import java.util.List;
+import java.util.Map;
 
 import reposense.model.Author;
+import reposense.model.Format;
 
 /**
  * Stores the contribution summary of the authors in the repo.
  */
 public class AuthorshipSummary {
     private final List<FileResult> fileResults;
-    private final HashMap<Author, Integer> authorFinalContributionMap;
+    private final Map<Author, Integer> authorFinalContributionMap;
+    private final Map<Author, Map<String, Integer>> authorFileFormatContributionMap;
 
-    public AuthorshipSummary(List<FileResult> fileResults, List<Author> authors) {
+    public AuthorshipSummary(List<FileResult> fileResults, List<Author> authors, List<Format> formats) {
         this.fileResults = fileResults;
         authorFinalContributionMap = new HashMap<>();
+        authorFileFormatContributionMap = new HashMap<>();
 
         // initialise each author contribution to be 0
         authors.forEach((author) -> authorFinalContributionMap.put(author, 0));
+        authors.forEach((author) -> {
+            Map<String, Integer> defaultFileFormatContribution = new LinkedHashMap<>();
+            for (Format format : formats) {
+                defaultFileFormatContribution.put(format.toString(), 0);
+            }
+            authorFileFormatContributionMap.put(author, defaultFileFormatContribution);
+        });
     }
 
     /**
-     * Increments the contribution count of {@code author} by one.
+     * Increments the contribution count of {@code author} and of a specific fileFormat by one.
```

Comment:
~and of a specific fileFormat by one.~ `and the corresponding file format specified by {@code filePath} by one.`